tests/test_transforms/test_timestamp/test_event_transform.py

```diff
@@ -0,0 +1,161 @@
+    df_exog.rename(columns={"target": "holiday"}, inplace=True)
+    df_exog["holiday"] = (
+        [0, 1, 0, 1, 1, 0, 0, 0, 1, 0] + [0, 0, 0, 0, 0, 0, 0, 0, 0, 0] + [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+    )
+    df = TSDataset.to_dataset(df)
+    df_exog = TSDataset.to_dataset(df_exog)
+    tsds = TSDataset(df, freq="D", df_exog=df_exog, known_future="all")
+    return tsds
+
+
+@pytest.mark.parametrize(
+    "mode,n_pre,n_post,ts_expected",
+    [
+        ("binary", 2, 2, "ts_check_event_transform_expected_1"),
+        ("binary", 1, 1, "ts_check_event_transform_expected_2"),
+        ("distance", 3, 3, "ts_check_event_transform_expected_3"),
+    ],
+)
+def test_fit_transform(ts_check_event_transform: TSDataset, ts_expected: TSDataset, mode, n_pre, n_post, request):
+    """Check event transform correctly generate holiday features"""
+    ts_expected = request.getfixturevalue(ts_expected)
+    transform = EventTransform(in_column="holiday", out_column="holiday", n_pre=n_pre, n_post=n_post, mode=mode)
+    result_binary = transform.fit_transform(ts=ts_check_event_transform).to_pandas()
+    pd.testing.assert_frame_equal(ts_expected.to_pandas(), result_binary)
+
+
+def test_in(ts_check_event_transform: TSDataset):
+    """Check that Exception raises when input column is not binary"""
+    transform = EventTransform(in_column="holiday", out_column="holiday", n_pre=1, n_post=1)
+    transform.fit_transform(ts_check_event_transform)
+    print(ts_check_event_transform)
+    print(ts_check_event_transform.df_exog)
+    print(ts_check_event_transform.regressors)
+
+
+def test_input_column_not_binary(example_tsds: TSDataset):
+    """Check that Exception raises when input column is not binary"""
+    transform = EventTransform(in_column="target", out_column="holiday", n_pre=1, n_post=1)
+    with pytest.raises(ValueError, match="Input columns must be binary"):
+        transform.fit_transform(example_tsds)
+
+
+@pytest.mark.parametrize("mode", ["bin", "distanced"])
+def test_wrong_mode_type(mode):
+    """Check that Exception raises when passed wrong mode type"""
+    with pytest.raises(NotImplementedError, match=f"{mode} is not a valid ImputerMode."):
+        _ = EventTransform(in_column="holiday", out_column="holiday", n_pre=1, n_post=1, mode=mode)
+
+
+@pytest.mark.parametrize("n_pre,n_post", [(0, 2), (-1, -1)])
+def test_wrong_distance_values(n_pre, n_post):
+    """Check that Exception raises when passed wrong n_pre and n_post parameters"""
+    with pytest.raises(ValueError, match=f"`n_pre` and `n_post` must be greater than zero, given {n_pre} and {n_post}"):
+        _ = EventTransform(in_column="holiday", out_column="holiday", n_pre=n_pre, n_post=n_post)
+
+
+def test_pipeline_with_event_transform(ts_with_binary_exog: TSDataset):
+    """Check that pipeline executes without errors"""
+    model = LinearPerSegmentModel()
+    event = EventTransform(in_column="holiday", out_column="holiday", n_pre=1, n_post=1)
```

Comment:
Here and in the next test better to remove "holiday" column from the dataset via `FilterTransform` to make sure that model works with generated features

Reply:
May be you can also add some test cases via parametrization